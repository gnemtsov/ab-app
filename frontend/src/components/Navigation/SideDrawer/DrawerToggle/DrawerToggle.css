.DrawerToggle {
    width: 2.4rem;
    height: 2rem;
    display: flex;
    flex-flow: column;
    justify-content: space-around;
    align-items: center;
    cursor: pointer;
}

.DrawerToggle div {
    width: 90%;
    height: 3px;
    background-color: white;
}

@media (min-width: 500px) {
    .DrawerToggle {
        display: none;
    }
}