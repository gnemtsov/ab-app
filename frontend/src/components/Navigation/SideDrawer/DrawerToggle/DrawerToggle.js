import React from 'react';

import classes from './DrawerToggle.css';

const drawerToggle = (props) => (
    <div className={classes.DrawerToggle} onClick={props.clicked}>
        <div></div>
        <div></div>
        <div></div>
    </div>
);

export default drawerToggle;