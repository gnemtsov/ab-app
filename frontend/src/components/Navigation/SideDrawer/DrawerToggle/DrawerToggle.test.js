import React from 'react';

import { configure, render, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import DrawerToggle from './DrawerToggle';

configure({ adapter: new Adapter() });

describe('DrawerToggle', () => {
	it('Basic render test', () => {
		const wrapper = shallow (
	        <DrawerToggle />
		);
		
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Click', () => {
		const mockClicked = jest.fn(() => {});
		
		const wrapper = shallow(
			<DrawerToggle
				clicked={mockClicked} />
		);
		
		wrapper.simulate('click');
		
		expect(mockClicked.mock.calls.length).toBe(1);
	})
});
