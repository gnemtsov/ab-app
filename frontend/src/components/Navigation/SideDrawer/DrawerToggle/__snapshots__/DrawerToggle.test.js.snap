// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerToggle Basic render test 1`] = `
<div
  className="DrawerToggle"
>
  <div />
  <div />
  <div />
</div>
`;
